### PR TITLE
Decouple store

### PR DESCRIPTION
**Pull request by DaveJarvis**
Apologies for the mammoth PR. Conceptually, this change pulls out the database items into their own **LoboStore** project. All the other changes are knock-on effects from addressing the dependencies. Broadly, these include:

* Introduce `ITabbedPane` interface to remove `DnDTabbedPane` from **LoboApi**. The `DnDTabbedPane` has a dependency on the database, but **LoboApi** need not have such coupling. The interface hides implementation-specific behaviours by referencing general Swing classes.
* Remove database references from `ColorFactory`. Similar to **LoboApi**, **LoboCommon** must avoid coupling to the database to prevent imposing such constraints on other developers without necessity. Caching behaviour is maintained to avoid polluting the colour map across web page requests.
* Decouple `FontFactory` from `LAFSettings`. The `LAFSettings` derives its values from the data store, but the `FontFactory` is part of **LoboCommon**. Provides a default font using a constant value, instead.
* Decouple `FontKey` from `LAFSettings`.
* Add `FontType` values that were from `LAFType`. This is because `LAFType` is used by **LoboJTattoo**, which has a dependency on the store.
* Remove `HttpNetwork`'s coupling to the user agent string found in the store. This would be better as a configuration setting assigned by developers. Updating the API in the future would allow setting the value from the UI, while still keeping the source of the data agnostic with respect to the store. Hard-coded for now, a constructor parameter is preferable.
* Update `StorageManager` to no longer load SQL for colours and user agent.

Nearly all the remaining changes result in decoupling the store facilities from the rendering facilities. In effect:

* **LoboStore** -- Depends on **LoboCommon**; all storage-related machinery goes here.
* **LoboEvo** -- Depends on **LoboStore**.
* **LoboHTML** -- Depends on **LoboStore**; these are now easier to eliminate (e.g., using common interfaces).
* **LoboJTattoo** -- Depends on **LoboStore**.

There are a few remaining dependencies in **LoboHTML** on **LoboStore** that can be removed.

### CODE REVIEW

**Comment by DaveJarvis** [comment]
Most of the issues are with magic numbers and minor formatting, neither of which should block PR. If certain formatting requirements are to be made, the build and commit system should perform them automatically. (I'll not change my IDE's autoformatting settings: every project is different. Similarly, my IDE is set to wildcard the import statements.)

Please consider relaxing the rules that block this PR.

**Comment by oswetto** [comment]
@DaveJarvis Thanks for your work. I like to create LoboStore to add the whole store mechanism.

From the Tools/Preferences/Look & Feels tab it is possible to set the color and font of the text and
from Tools/Preferences/General tab it is possible to set user agent. 

With your fixes these features no longer work.
Without removing the data from the database we can check if the database is present, if it is not present we can implement the default values ​​for the objects that use the archive data.

File: LoboStore/src/main/java/com/loboevolution/store/laf/LAFSettings.java [hunk 1]
```diff
@@ -190,7 +190,7 @@ private LAFSettings retriveFontDate() {
 				laf.setStrikethrough(rs.getInt(18) == 1);
 				laf.setFontSize(Float.parseFloat(rs.getString(19)));
 				laf.setFont(rs.getString(20));
-				laf.setColor(ColorFactory.getInstance().getColor(rs.getString(21)));
+				laf.setColor(Color.BLACK);
 				laf.setBold(rs.getInt(22) == 1);
```
Comment (oswetto) [inline LoboStore/src/main/java/com/loboevolution/store/laf/LAFSettings.java:194]: From the Tools/Preferences/Look & Feels tab it is possible to set the text color, with this fix is not more possible.

**Review by oswetto — CHANGES_REQUESTED** [review]

**Comment by DaveJarvis** [comment]
> From the Tools/Preferences/Look & Feels tab it is possible to set the text color, with this fix is not more possible.

I'm not going to address each of the points, because they are nearly all the same issue.

The code, as it stands, unnecessarily couples the data store to HTML rendering. For applications that need only HTML rendering, without the additional complexity necessary for a web browser and user preferences, this is an unfortunate decision that arbitrarily limits the potential without strong justification, IMO.

I completely understand that the functionality I've removed will cause breaking changes to the LoboEvo browser. You may wish to consider exposing the parameters another way, one that doesn't involve direct access to the LoboStore (i.e., SQL database).

Take the User Agent String, for example. Rather than derive its value from the data store, the LoboEvo browser could configure the User Agent value as a String. When LoboEvo browser runs, it reads from the data store, then sets the User Agent accordingly. This same approach can be made for each of the issues mentioned in the PR.

Feel free to close the PR.

**Pull request merged**